src/librustc/infer/error_reporting/need_type_info.rs

```diff
@@ -334,6 +375,18 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
                 format!("consider giving this pattern {}", suffix)
             };
             err.span_label(pattern.span, msg);
+        } else if let Some(ExprKind::MethodCall(segment, ..)) = local_visitor.found_method_call {
+            if let Ok(snippet) = self.tcx.sess.source_map().span_to_snippet(segment.ident.span) {
+                if segment.args.is_none() {
+                    err.span_suggestion(
+                        segment.ident.span,
+                        "consider specifying the type argument in the method call",
+                        // FIXME: we don't know how many type arguments should be set here.
+                        format!("{}::<_>", snippet),
```

Comment:
@nikomatsakis I'd need some help here to make sure the suggestion we provide is actually useful instead of potentially misleading. How could I gather the `DefId` of the called method? Once I've got that I could easily verify if it _has_ type arguments, and if it matches the one we couldn't resolve.